### PR TITLE
Move over hamaddr, set up workspace, other misc changes.

### PR DESCRIPTION
**Pull request by darconeous**
I also made a few changes to hamaddr. I'll be removing the one in `arnce-spec` so that this one becomes authoritative.

### CODE REVIEW

File: hamaddr/src/error.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+use std::fmt;
+use std::fmt::{Debug, Display, Formatter};
+
+pub type Result<T = (), E = anyhow::Error> = std::result::Result<T, E>;
+
+/// Error type indicating an invalid character at a specific index.
+#[derive(Debug, thiserror::Error)]
+pub struct InvalidCharAt(pub usize);
+
+impl Display for InvalidCharAt {
```
Comment (dancrossnyc) [inline hamaddr/src/error.rs:10]: Q: can you derive Display instead of using an explicit impl?

File: hamaddr/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+name = "hamaddr"
+version = "0.1.0"
+edition = "2021"
+authors = ["kz2x", "n6drc"]
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+anyhow = "1.0"
+thiserror = "1.0"
```
Comment (dancrossnyc) [inline hamaddr/Cargo.toml:11]: Add newline at end of file.

**Review by dancrossnyc — APPROVED** [review]
The hamaddr change almost triples the size of the crate, and I find the chunked representation harder to understand than the u64 it replaces. The new functionality is great, but it feels like complexity has intruded to an undue extent. That said, it's there and the tests pass, so we should use it.

File: hamaddr/src/error.rs (hunk 1 above)
Comment (darconeous) [reply hamaddr/src/error.rs:10]: Nope, `Display` can't be derrived.

Comment (dancrossnyc) [reply hamaddr/src/error.rs:10]: Ah, I was thinking of derive_more::Display.

**Comment by darconeous** [comment]
I think the trippling of the size of the crate is largely due to added features, checks, and documentation. The underlying storage was converted to `[u8;8]` for improved convenience, since it is the same order that packets are in. The chunked representation is just for conversions. My brain also thinks in chunks, although I've learned that what seems most straightforward to one person can easily be obtuse to another. With another pass or two (to remove dependence on `anyhow::Error`, specifically) I can make the crate largely usable in `nostd` environments, which seems cool.

I'll take another pass at simplification before landing, to remove unused parts and to add some additional documentation.

**Comment by darconeous** [comment]
Oh, haha, I see you already landed it. I'll add another PR for the changes.

**Pull request merged**